### ISSUE TITLE
Sync between devices

### ISSUE THREAD

**Issue by themaxw** [matthiasemde/musikus-android#76]
It would be great if i was able to sync my practice sessions between devices, for some instruments i like to use a different device while practicing.
Sync could (maybe) be implemented via an external webDAV server. 
Is this in the realm of possibilities?

**Comment by matthiasemde** [comment]
Hi @themaxw,
while syncing between devices is a feature we would like to add to the app very much, we would need to find a scalable and preferably free way to implement it. Handling user data online also brings with it a whole other can of worms in the form of user authentication and data privacy.
To answer your question: Yes, data sync is surely in the real of possibilities and we would like it as a feature, but I do not see it on our roadmap until further down the line.
Best whishes
Matthias

**Comment by mipro98** [comment]
Hello @themaxw,

as Matthias already said, a natively built-in sync solution might not be so easy regarding privacy and other constraints for us (an independent small open-source project with limited capacity). It might be relatively easy (and also cheap/free to an extent) to implement this feature e.g. using a Firebase database but this makes us also dependent on Play Services and Google dependencies.

However, what *might* be feasible hook into existing data syncing solutions the user already uses like your mentioned WebDav solution (which is e.g. backed by the user's own Nextcloud) and we don't really store data, i.e. you still *own* your data, we only provide an API for syncing.

This is definitely an awesome feature and huge enabler for many people practicing on multiple devices and also opens the door to many other possiblities (third-party clients / web-client etc.) but since the scope is quite large and the effort pretty high to implement this will only be on our roadmap when the app has a lot more maturity and  all current bugs/issues addressed.

**Comment by alworx** [comment]
I agree with @mipro98, a proprietary/built-in solution is a lot of effort and most probably costly in the long term.

Shouldn't [SAF](https://developer.android.com/guide/topics/providers/document-provider) cover that functionality if the current database in stored externally?
Example client: https://github.com/android/storage-samples/tree/main/StorageClient
